src/parser.y

```diff
@@ -661,10 +662,12 @@ void yyerror (char const *s)
          (*sourcefile()) % lexerget_lineno() % s);
 }
 
-bool parse(FileModule *&module, const char *text, const std::string &filename, int debug)
+bool parse(FileModule *&module, const char *text, const std::string &filename, int debug, bool mainModule)
 {
   fs::path path = fs::absolute(fs::path(filename));
   
+  bMainModule = mainModule;
```

Comment:
Would it be cleaner to instead pass the `assignmentWarning` flag as an argument, to have one less piece of logic to manage?